clickhouse-http-client/src/main/java/com/clickhouse/client/http/ClickHouseHttpClient.java

```diff
@@ -28,6 +30,57 @@ public class ClickHouseHttpClient extends AbstractClient<ClickHouseHttpConnectio
 
     static final List<ClickHouseProtocol> SUPPORTED = Collections.singletonList(ClickHouseProtocol.HTTP);
 
+    public static class HostNameAndAddress{
+        public String hostName;
+        public String address;
+    }
+
+    private static HostNameAndAddress getLocalHost() {
+        // get local address but not localhost
+        HostNameAndAddress hostNameAndAddress = new HostNameAndAddress();
+        try {
+            Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
```

Comment:
This is the only place where SocketException may be throw and only in case when host has no interface. 
It is a good practice to minimize scope of a try-catch to deal only with a few exceptions at a time.
Please, rethrow SocketException inside a RuntimeException to indicate a critical error.